### DISCUSSION TITLE
App content not rendering on Ubuntu

### DISCUSSION THREAD

**Discussion by BrunoDenisov** [AvaloniaUI/Avalonia#17061] [Q&A] [answered]
I was my developing the app on windows and wanted to make is linux (or at the least ubuntu compatible), I have dual boot and switched over, but when I try to run or debug the app, the window it self starts up, but any content on the inside of the window does not render. I have a content control in my main window that upon start up navigates to a home view.  If anyone experienced a similar issue before, I would greatly appreciate any help or ideas on how to fix it. 

![image](https://github.com/user-attachments/assets/b6b3ddc2-e2ac-443c-9126-fca7d2e2dc31)

It look like this when the app is launched. You can "see through it".

**Comment by BrunoDenisov** [comment]
I have checked #4826 and attempted that, but it does not seem like a monitor scaling issues

**Comment by BrunoDenisov** [comment]
Further info, The app seems to be locked up, I can't exit it through pressing X, resizing the "windwo" works. I'm investigating is is stuck on some task.

**Reply by thevortexcloud** [reply]
As a sanity check, it's probably worth checking if a fresh Avalonia app built directly on Linux works. EG

```bash
dotnet new install Avalonia.Templates
dotnet new avalonia.app
dotnet run
```

**Reply by BrunoDenisov** [reply]
This is fair, and I've checked. 
What I've figgured out I'm accesing TOP level in app.axaml.cs and assining it staticlly so I can access it with just App.Toplevel, and when I comment  TopLevel = TopLevel.GetTopLevel(desktop.MainWindow); this line of code, the App renders fine, so I've got to figure out a way do access Top level another way or figure out what is cauing top level to break my app.

**Reply by thevortexcloud** [reply]
Why do you need a static top level? I can't think of a single reason to do this if you have followed MVVM properly.

**Reply by BrunoDenisov** [reply]
I don't need it there its just how I got it to works, Its my first more comlicated app then just a single view and view model doing something. I am aware how ever that it is no the best approach to doing it.

**Reply by thevortexcloud** [reply]
It does not make much sense to me for it to be a static as you may have multiple top levels. You should only try to get the top level from the view you need it in. View models should never access top levels directly. EG https://github.com/AvaloniaUI/Avalonia/discussions/13484#discussioncomment-7478892

I would suggest trying to refactor your code if you can. That said I am not sure why doing what you are doing would stop the app from rendering properly.

**Reply by BrunoDenisov** [reply]
It is fine I can refactore my app, what the app id doing, It a app that generates a json based on parameters selectd on the app and then later it calls a C binary to build a sort of update file. When the app loads what its using top level for initially is loading files that can be selected to that their paths can be put into the JSON. Why I'm confused the app work fine for Windows, but as soon as I'm on Ubuntu it breaks. But thats for the help

**Reply by BrunoDenisov** [reply]
Do you maybe have an example on how do do it porperly, I'm sturgling to find a good resource.

**Reply by thevortexcloud** [reply]
Are you using ReactiveUI, CommunityMVVM or something else?

**Reply by BrunoDenisov** [reply]
CommunityMVVM currently

**Reply by thevortexcloud** [reply, marked as answer]
https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/ViewInteraction/DialogManagerSample

There are other ways of doing it besides that, but I don't have any other examples on hand for CommunityMVVM.

EDIT: Just noticed this one

https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/ViewInteraction/MvvmDialogSample#solution-2--write-your-own-interaction-class
